ethcore/src/machine.rs

```diff
@@ -193,6 +193,26 @@ impl EthereumMachine {
 						.and_then(|b| state.transfer_balance(child, beneficiary, &b, CleanupMode::NoEmpty))?;
 				}
 			}
+
+			if let Some(irregular_changes) = self.params.irregular_state_changes.get(&block.header().number()) {
```

Comment:
This block does not depend on `ethash_params`, does it have to be inside that particular `if`? If yes could you please add a comment explaining that.

Reply:
definitely should be moved outside. other engines might have irregular state changes too.